fog/ledger/test_infra/src/lib.rs

```diff
@@ -286,3 +295,67 @@ impl Ledger for MockLedger {
         unimplemented!()
     }
 }
+
+pub struct ShardProxyServer {
+    server_handle: JoinHandle<Result<(), hyper::Error>>,
+    stop_channel: oneshot::Sender<()>,
+}
+
+impl ShardProxyServer {
```

Comment:
I don't think these tests are providing any assurances about the behavior of a shard with multiple stores. Thus I think this is adding bunch of complexity to the tests with little to no assurances of behavior.

These tests look like they are testing
```mermaid
flowchart  LR
A(Router) --> B(Shard 1)
B --> F[Store A]
B --> G[Store B]
B --> H[Store C]
```
but this `ShardProxyServer` is faking out the shards, so these tests are
```mermaid
flowchart  LR
A(Router) --> B(Fake Shard 1)
B --> F[Store A]
B --> G[Store B]
B --> H[Store C]
```
Generally in testing one uses a test double (fake, mock, stub, etc) at the edges of what's being tested, but this is in the middle of what's being tested.

Reply:
This was discussed offline. The purpose of the test is to test how the Router handles getting replies back from multiple shards. The shard proxy aren't tested directly as they will be implementation specific to the deployment.